draft-ietf-tls-dtls-connection-id.md

```diff
@@ -138,10 +138,11 @@ receive as a CID in encrypted records, it is possible
 for an endpoint to use a globally constant length for such connection
 identifiers.  This can in turn ease parsing and connection lookup,
 for example by having the length in question be a compile-time constant.
-Implementations, which want to use variable-length CIDs, are responsible
+Such implementations must still be able to send
+CIDs of different length to other parties.
```

Comment:
That's an interesting point. some weeks ago I considered to raise an issue for that. Generally constraint peers don't limit the CID length just for fun. They are not able to use larger ones. The upper limit of 255 bytes (hope I'm right) is more academical value. I guess, in constraint deployments 32 bytes will be mostly enough. My conclusion was therefore, that a server just documents his max. CID length. If more flexibility is really required, but constraint devices should be supported, I guess that max cid-length must be negotiated.

Reply:
The intent here was just to improve the readability of the transition from "when/how you can use fixed CID length" to "considerations about variable-length CIDs"; the actual technical requirements are specified elsewhere in the document.
(I think I also asked a question about whether there would be reason to lower the 255-byte limit, in the emailed review comments, which might be a better place to discuss this topic.)

It seems that the scenario where a node can only handle max (say) 32 byte CID is asymmetric if that node is client or server.  If it's the server, it has the option of rejecting CID entirely if the client uses a longer CID than it supports, but the client would have to do a fallback dance (retry without the CID extension) if the server said to use a "too-long" CID.  I was reluctant to suggest a very small cap on CID length since (IIRC) some people wanted to use the CID value as a self-encrypted blob to store routing information/etc.

Reply:
FWIW, QUIC caps it at 20 octets, but does not allow the peer to indicate a restriction.

Reply:
I am fine with the editorial change. As there is no desire to change implementation of the mechanism or to add new requirements I believe there is not much more to do with this item.

Reply:
I think there is a normative MUST here.

Reply:
MUST language added.